File: DAY 9/solutions.java/1456. Maximum Number of Vowels in a Substring of Given Length.java
import java.util.*;

class Solution {
    public int maxVowels(String s, int k) {

        int left = 0;
        int count = 0;
        int answer = 0;

        Set<Character> vowels = new HashSet<>();
        vowels.add('a');
        vowels.add('e');
        vowels.add('i');
        vowels.add('o');
        vowels.add('u');

        for (int right = 0; right < s.length(); right++) {

            if (vowels.contains(s.charAt(right))) {
                count++;
            }

            if (right - left + 1 == k) {

                answer = Math.max(answer, count);

                if (vowels.contains(s.charAt(left))) {
                    count--;
                }

                left++;
            }
        }

        return answer;
    }
}

File: DAY 9/solutions.py/1456. Maximum Number of Vowels in a Substring of Given Length.py
class Solution:
    def maxVowels(self, s: str, k: int) -> int:
        left =0 
        count=0
        answer=0
        vowels = set("aeiou")
        for right in range(len(s)):
            if s[right] in vowels:
                count+=1
            if right - left +1==k:  
                answer=max(answer,count)
                if s[left] in vowels :
                    count-=1
                left+=1
        return answer            
